src/NUnitFramework/framework/TestParameters.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace NUnit.Framework
+{
+    /// <summary>
+    /// TestParameters class holds any named parameters supplied to the test run
+    /// </summary>
+    public class TestParameters
+    {
+        private readonly Dictionary<string, string> _parameters = new Dictionary<string, string>();
+
+        /// <summary>
+        /// Gets the number of test parameters
+        /// </summary>
+        public int Count
+        {
+            get { return _parameters.Count; }
+        }
+
+        /// <summary>
+        /// Gets a collection of the test parameter names
+        /// </summary>
+        public ICollection<string> Names
+        {
+            get { return _parameters.Keys; }
+        }
+
+        /// <summary>
+        /// Indexer is the only public access to the internal dictionary
+        /// </summary>
+        /// <param name="name">Name of the parameter</param>
+        /// <returns>Value of the parameter or null if not present</returns>
+        public string this[string name]
+        {
+            get { return _parameters.ContainsKey(name) ? _parameters[name] : null; }
+        }
+
+        /// <summary>
+        /// Get method is a simple alternative to the indexer
+        /// </summary>
+        /// <param name="name">Name of the paramter</param>
+        /// <returns>Value of the paramter or null if not present</returns>
+        public string Get(string name)
+        {
+            return _parameters.ContainsKey(name) ? _parameters[name] : null;
+        }
+
+        public string Get(string name, string defaultValue)
+        {
+            return _parameters.ContainsKey(name) ? _parameters[name] : defaultValue;
+        }
+        
+        public int Get(string name, int defaultValue)
```

Comment:
Fair enough.
On Jun 15, 2016 9:16 PM, "Neil Colvin" <notifications@github.com> wrote:

> In src/NUnitFramework/framework/TestParameters.cs
> <https://github.com/nunit/nunit/pull/1570#discussion_r67285201>:
>
> > +        /// <summary>
> > +        /// Get method is a simple alternative to the indexer
> > +        /// </summary>
> > +        /// <param name="name">Name of the paramter</param>
> > +        /// <returns>Value of the paramter or null if not present</returns>
> > +        public string Get(string name)
> > +        {
> > +            return _parameters.ContainsKey(name) ? _parameters[name] : null;
> > +        }
> > +
> > +        public string Get(string name, string defaultValue)
> > +        {
> > +            return _parameters.ContainsKey(name) ? _parameters[name] : defaultValue;
> > +        }
> > +
> > +        public int Get(string name, int defaultValue)
>
> It is sometimes easier and cleaner to have an if (..Exists ("abc")) then
> an if (...Get("abc") = null).
>
> Also, it generates less code.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/nunit/nunit/pull/1570/files/1524aef16283bd47073743e47bc6e91883bff22e#r67285201>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACjfhYD8fNDZX7bFuhD92B46hgy2nuluks5qMM4hgaJpZM4I0238>
> .
>